src/VisualStudio/IntegrationTest/IntegrationTests/AbstractInteractiveWindowTest.cs

```diff
@@ -112,7 +129,37 @@ protected void VerifyReplPromptConsistency(string prompt, string output)
             }
         }
 
+        protected void VerifyCompletionItemExists(params string[] expectedItems)
+        {
+            var completionItems = InteractiveWindow.GetCompletionItems();
+            foreach (var expectedItem in expectedItems)
+            {
+                Assert.Contains(expectedItem, completionItems);
+            }
+        }
+
+        public void VerifyCurrentTokenType(string tokenType)
+        {
+            WaitForAsyncOperations(
+                FeatureAttribute.SolutionCrawler,
+                FeatureAttribute.DiagnosticService,
+                FeatureAttribute.Classification);
+            var actualTokenTypes = InteractiveWindow.GetCurrentClassifications();
+            Assert.Equal(actualTokenTypes.Length, 1);
+            Assert.Contains(tokenType, actualTokenTypes[0]);
+            Assert.NotEqual("text", tokenType);
+        }
+
+        protected void InvokeCompletionList()
+        {
+            ExecuteCommand(WellKnownCommandNames.Edit_ListMembers);
+            WaitForAsyncOperations(FeatureAttribute.CompletionSet);
```

Comment:
It's most fishy to me this isn't waiting for the Workspace operation before invoking the command. Otherwise we might be getting partial results. @tannergooding or maybe @dpoeschl did we guard against that somewhere?